library/entropy.c

```diff
@@ -466,28 +466,27 @@ int mbedtls_entropy_update_nv_seed( mbedtls_entropy_context *ctx )
 #if defined(MBEDTLS_FS_IO)
 int mbedtls_entropy_write_seed_file( mbedtls_entropy_context *ctx, const char *path )
 {
-    int ret = MBEDTLS_ERR_ENTROPY_FILE_IO_ERROR;
-    FILE *f;
+    int ret;
```

Comment:
It has a little significance here, but for consistency we normally initialise the return value to `MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED` .

Reply:
Thanks for pointing out; will fix.

Does this rule hold for all variables or for return values only? I see that some stack-allocated stuff does not get initialized at the point of declaration that's why I omitted ret initialization since it gets overwritten anyway at ln.473. I red the coding standards but did not find the corresponding requirements (or I just missed them).

Reply:
However, in case if I change the code as suggested below there is no need to initialize ret to `MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED`; it will be better to have 0 here so if all three file-related calls succeed ret will be equal to 0 or gets updated to a required err. code otherwise. We have exactly this approach below in `mbedtls_entropy_update_seed_file` where `MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED` is not used. Or this is a kind of marker for stack examination in case of crash?

Reply:
It just holds for return values and it is not a hard rule. I don't think this is mentioned in the coding standard, but you are right, maybe it should be.

Initialising to `MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED` is not really necessary, it is there mostly for robustness and for the principle of failing by default. If we make a mistake in the function, it is less likely that we return success.

This goal would be still achieved if `ret` was initialised to `MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED` and I think that it would be slightly better like that. But this is a minor issue and I think we can merge the PR even if you don't feel like fixing this.

Reply:
@yanesca thanks for the explanation. I'll rework this part ASAP and will push a new commit. I believe this should be included into the code standard since this is a kind of standard prologue for each (at least public) API component which acts like a guard to prevent errors propagation and thus this approach should be enforced by default (for the folks like me not to make the same mistake in future).

Reply:
I agree, I have added a remark about this to a corresponding issue:
https://github.com/ARMmbed/mbedtls/issues/3636#issuecomment-686335969